cardinal/search/searchcomponent.go

```diff
@@ -0,0 +1,25 @@
+package search
+
+import "pkg.world.dev/world-engine/cardinal/types"
+
+// This file represents primitives for search
+// The primitive wraps the component and is used in search
+// the purpose of the wrapper is to prevent the user from ever
+// instantiating the component. These wrappers are used to check if entities
+// contain the specified component during the search.
+
+type componentWrapper struct {
+	Component types.Component
+}
```

Comment:
any reason we can't just return a `types.Component` from the function below? not sure i understand why we need the wrapper

Reply:
you'll get a compile error if you try to pass in a struct value to the generic parameter though

this won't compile:
```
cardinal.NewSearch(wCtx).Contains(search.Component[EnergyComponent{Amt: 3}]())
```

Reply:
oh wait i think i see why now

Reply:
nevermind lol